src/execution/vm/bytecode_generator.cpp

```diff
@@ -1887,6 +1887,100 @@ void BytecodeGenerator::VisitBuiltinArithmeticCall(ast::CallExpr *call, ast::Bui
   GetExecutionResult()->SetDestination(dest);
 }
 
+void BytecodeGenerator::VisitBuiltinAtomicAndCall(ast::CallExpr *call) {
+  auto args = call->Arguments();
```

Comment:
You're making a copy of the arguments vector here, you need to make this `auto &args`. Or, better yet, make it `const auto &args`.